src/figuresClasses.ts

```diff
@@ -1,11 +1,111 @@
-export interface Figure {}
+// export interface Figure {}
 
-export class Triangle implements Figure {}
+// export class Triangle implements Figure {}
 
-export class Circle implements Figure {}
+// export class Circle implements Figure {}
 
-export class Rectangle implements Figure {}
+// export class Rectangle implements Figure {}
 
-export function getInfo(figure): string {
-  return typeof figure;
+// export function getInfo(figure): string {
+//   return typeof figure;
+// }
+
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, y) => x - y);
+
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. The error message 'Sides a, b, and c can't form a triangle' is good, but it would be even better if it included the reason, such as 'The sum of any two sides must be greater than the third side.'